#! /bin/sh /usr/share/dpatch/dpatch-run

## 21_vdr-1.3.31-skipframes.dpatch by Reinhard Nissl <rnissl@gmx.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes a problem when moving cutting marks in radio recordings

@DPATCH@
diff -urNad vdr-1.3.31/dvbplayer.c /tmp/dpep.pCvmik/vdr-1.3.31/dvbplayer.c
--- vdr-1.3.31/dvbplayer.c	2005-08-14 12:52:45.000000000 +0200
+++ /tmp/dpep.pCvmik/vdr-1.3.31/dvbplayer.c	2005-08-31 21:27:50.253617352 +0200
@@ -621,7 +621,10 @@
      int Current, Total;
      GetIndex(Current, Total, true);
      int OldCurrent = Current;
-     Current = index->GetNextIFrame(Current + Frames, Frames > 0);
+     // As GetNextIFrame() increments/decrements at least once, the 
+     // destination frame (= Current + Frames) must be adjusted by
+     // -1/+1 respectively.
+     Current = index->GetNextIFrame(Current + Frames + (Frames > 0 ? -1 : 1), Frames > 0);
      return Current >= 0 ? Current : OldCurrent;
      }
   return -1;
